contracts/oracles/chainlink/ChainlinkUSDMultiOracle.sol

```diff
@@ -104,3 +105,57 @@ contract ChainlinkUSDMultiOracle is IOracle, AccessControl, Constants {
         updateTime = (updateTime1 < updateTime2) ? updateTime1 : updateTime2;
     }
 }
+
+/**
+ * @title ChainlinkL2USDMultiOracle: ChainlinkUSDMultiOracle that's safe to use on L2
+ * @notice Chainlink recommends checking the sequencer status on some L2 networks to avoid
+  reading stale data
+  
+  https://docs.chain.link/docs/l2-sequencer-flag/
+ */
+contract ChainlinkL2USDMultiOracle is ChainlinkUSDMultiOracle {
+    using CastBytes32Bytes6 for bytes32;
+
+    FlagsInterface public chainlinkFlags;
+    // https://docs.chain.link/docs/l2-sequencer-flag/
+    address internal constant FLAG_ARBITRUM_SEQ_OFFLINE =
+        address(bytes20(bytes32(uint256(keccak256('chainlink.flags.arbitrum-seq-offline')) - 1)));
+
+    constructor(FlagsInterface flags) {
+        require(address(flags) != address(0), 'FlagsInterface has to be set');
+        chainlinkFlags = flags;
+    }
+
+    function assertCanGetPrice() private view {
```

Comment:
This looks like a good candidate for a modifier:
```
modifier onlyFresh() {
  // If flag is raised we shouldn't perform any critical operations
  require(
    chainlinkFlags.getFlag(FLAG_ARBITRUM_SEQ_OFFLINE) != isRaised,
    "Chainlink feeds are not being updated"
  );
  _;
}
```

Reply:
I find modifiers easy to miss, so went for an explicit check in the function body
Also, remember Tenderly always pointing you at `auth` modifier as the source of the revert? That's another problem with modifiers

With that said, I'm not super opposed to them. Took your suggestion